wrappers/python/nntile/layer/linear.py

```diff
@@ -304,3 +305,42 @@ def from_torch(torch_linear, x, hidden_dim_tile, redux, next_tag):
             linear_nntile.b.value.from_array(torch_linear.bias.data.cpu().detach().numpy())
 
         return linear_nntile, next_tag
+
+    def get_layer_forward_flops(self):
+        x_shape = self.x.value.shape
+        w_shape = self.w.value.shape
+        if self.side == "L":
+            return 2 * np.prod(x_shape) * np.prod(w_shape[self.ndim:])
+        elif self.side == "R":
+            return 2 * np.prod(x_shape) * np.prod(w_shape[:-self.ndim])
+
+    def get_layer_backward_flops(self):
+        x_shape = self.x.value.shape
+        w_shape = self.w.value.shape
+        y_grad_shape = self.y.grad.shape
+        w_grad_flops = 0
+        x_grad_flops = 0
+        if self.side == "L":
+            if self.w.grad_required:
+                w_grad_flops = (2 * np.prod(x_shape) *
+                                np.prod(y_grad_shape[self.ndim:]))
+            if self.x.grad_required:
```

Comment:
amount of operations is the same both for `w.grad` and `x.grad`. No need to calculate it twice differently.

Reply:
Well, it is not obvious, the shapes of ``w`` and ``x`` may be different, as well as the prod of these shapes